### PR TITLE
refactor zookeeper config & make zookeeper unit-test work ok in hook checks

### PR DESCRIPTION
**Pull request by qinzuoyan**
- refactor zk config: https://github.com/imzhenyu/rDSN/issues/46, but there are still some places to improve, e.g., how to specify lock_lock of distributed_lock_service_zookeeper
- make zk unit-test work ok in hook checks: https://github.com/imzhenyu/rDSN/issues/42
- fix some rep_tests: https://github.com/imzhenyu/rDSN/issues/41, but there are still some cases to fix.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,10 +12,18 @@ before_install:
...
+
+before_script:
+    - ./start_zk_for_test.sh
+    - ulimit -c unlimited -S
 
 script:
-    - ./check_format.sh
-    - ./build.sh
-    #	&& ./test.sh
+    - ./test.sh
```
Comment (imzhenyu) [inline .travis.yml:25]: are u sure we are ready for this? I mean some of the tests may still not pass at this moment. Or they are all fixed already?:)

Comment (qinzuoyan) [inline .travis.yml:25]: Not ready now, I am fixing it.

Comment (qinzuoyan) [inline .travis.yml:25]: fixed

**Comment by imzhenyu** [comment]
my next pull request will have conflict with this. since Linux test is already ok, i'm going to merge this to avoid conflict and unblock the next pull request:)

**Comment by qinzuoyan** [comment]
This pull request is closed, but more in https://github.com/imzhenyu/rDSN/pull/67

**Pull request merged**